### PR TITLE
Stop using git submodules

### PR DESCRIPTION
**Pull request by quyykk**
This stops our use of a git submodule, in favor of a CMake script that simply downloads vcpkg if necessary. GitHub Desktop has issues with submodules, so that's why we have to stop using them.

## Testing Done

I have tested this locally and it works just fine. CI should additionally test it as well.

### CODE REVIEW

File: utils/vcpkg_bootstrap.cmake [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+        # Apply patches
+        foreach(_patch IN LISTS VCPKG_PATCHES)
+            message(STATUS "Applying patch ${_patch}")
+            execute_process(
+                COMMAND "${GIT_EXECUTABLE}" apply --whitespace nowarn "${_patch}"
+                COMMAND "${CMAKE_COMMAND}" -E touch "${X_VCPKG_CLONE_DIR}/${_patch_name}.applied"
+                WORKING_DIRECTORY "${X_VCPKG_CLONE_DIR}"
+                COMMAND_ERROR_IS_FATAL ANY
+            )
+        endforeach()
```
Comment (MCOfficer) [inline utils/vcpkg_bootstrap.cmake:97]: A lot of lines here could probably be safely removed for better maintainability. We don't have an vcpkg patches, so all lines concerning them are moot. Likewise for any registry type other than `builtin`.

File: CMakeLists.txt [hunk 2]
```diff
@@ -20,14 +21,16 @@ set(CMAKE_INTERPROCEDURAL_OPTIMIZATION_RELEASE TRUE)
 set(CMAKE_BUILD_RPATH_USE_ORIGIN ON)
 
 # Setup vcpkg.
-set(VCPKG_BOOTSTRAP_OPTIONS "-disableMetrics")
-set(CMAKE_POLICY_DEFAULT_CMP0077 NEW)
-set(X_VCPKG_APPLOCAL_DEPS_INSTALL ON)
-if(NOT EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/vcpkg/scripts/buildsystems/vcpkg.cmake" AND NOT DEFINED CMAKE_TOOLCHAIN_FILE)
-	message(FATAL_ERROR "Can't find vcpkg, did you forget to run 'git submodule update --init'?")
+if(NOT ES_NO_VCPKG)
```
Comment (MCOfficer) [inline CMakeLists.txt:24]: Probably better to invert `ES_NO_VCPKG` to `ES_USE_VCPKG`.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -555,6 +555,8 @@ jobs:
         sudo apt install -y --no-install-recommends libxmu-dev libxi-dev libgl-dev libglu1-mesa-dev libgles2-mesa-dev libwayland-dev libxkbcommon-dev libegl1-mesa-dev libosmesa6 mesa-utils libglvnd-dev
     - uses: lukka/get-cmake@latest
     - uses: lukka/run-vcpkg@v10
+      with:
```
Comment (petervdmeer) [inline .github/workflows/ci.yml:558]: The checkout a few lines above here says `submodules: true`. Is that still desired?

Comment (petervdmeer) [reply .github/workflows/ci.yml:558]: Also for other platforms...

**Review by MCOfficer — APPROVED** [review]

**Pull request merged**